lib/msal-node/src/network/HttpClientWithRetries.ts

```diff
@@ -22,41 +25,64 @@ export class HttpClientWithRetries implements INetworkModule {
         this.retryPolicy = retryPolicy;
     }
 
-    async sendGetRequestAsync<T>(
-        url: string,
-        options?: NetworkRequestOptions
-    ): Promise<NetworkResponse<T>> {
-        // the underlying network module (custom or HttpClient) will make the call
-        let response: NetworkResponse<T> =
-            await this.httpClientNoRetries.sendGetRequestAsync(url, options);
+    private retryAfterMillisecondsToSleep(
+        retryHeader: http.IncomingHttpHeaders["retry-after"]
+    ): number {
+        if (!retryHeader) {
+            return 0;
+        }
 
-        let currentRetry: number = 0;
-        while (
-            await this.retryPolicy.pauseForRetry(response.status, currentRetry)
-        ) {
-            response = await this.httpClientNoRetries.sendGetRequestAsync(
-                url,
-                options
+        // retry-after header is in seconds
+        let millisToSleep = Math.round(parseFloat(retryHeader) * 1000);
```

Comment:
Is it possible to move this logic inside the retry policy itself? The HttpClientWithRetries should onyl be responsible for waiting N ms and retrying. It should figure out N (how long to wait) from the retry policy itself.